cli/src/main/java/org/jboss/as/cli/impl/ExtensionsLoader.java

```diff
@@ -213,6 +210,12 @@ private void addError(String msg) {
         }
     }
 
+    private static String canonicalize(String moduleName) {
+        // This isn't a full canonicalization, but extension module names are unlikely to be non-canonical
+        // in any way beyond maybe a trailing ":main". So just check for and strip any trailing ":main"
```

Comment:
question: Did you consider adding a ModuleIdentifierUtil version to the wildfly-controller-client module so we can have the utility available on the client code?

Reply:
I hadn't no, but I updated this to move the main code to controller-client and have the controller module class just delegate to the client class.

I wouldn't want to do too much of this kind of thing, as its kind of an edge case for controller-client. But after almost 15 years we have hardly any of these, so I think it's ok. ;-)